csharp/src/Drivers/Apache/Spark/SparkConnection.cs

```diff
@@ -258,6 +258,7 @@ internal SparkConnection(IReadOnlyDictionary<string, string> properties)
 
         private void ValidateProperties()
         {
+            tlsOptions =  HiveServer2SslImpl.ValidateTlsOptions(Properties);
```

Comment:
Can we put this inside of `ValidateOptions`? I'd prefer we validate the Authentication/Connection first.